custom-pages/actions/Initial AD login Creates Account in Identity/code.js

```diff
@@ -0,0 +1,129 @@
+// @ts-nocheck
+/**
+* Handler that will be called during the execution of a PostLogin flow.
+*
+* @param {Event} event - Details about the user and the context in which they are logging in.
+* @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+*/
+
+exports.onExecutePostLogin = async (event, api) => {
+
+    //this rule is only concerned with migrating people who sign in for the first time with AD
+    //or google.
+    if ((event.connection.strategy !== "waad" && 
+         event.connection.strategy !== "google-oauth2")){
+      console.log(event.connection.strategy);
+      return;
+    } 
+  
+    const isInIdentityDB =  !!(event.user.user_metadata && event.user.user_metadata.isInIdentityDB);
+  
+    if (isInIdentityDB){
+      console.log('Initial AD Login creates account in Identity - User already in identity db');
+      return;
+    }
+    else{
+  
+      console.log('Initial AD Login creates account in Identity - Creating user in Identity db');
+      
+      var tokenResponse, tokenResponseBody
+  
+      const params = new URLSearchParams({
+        grant_type: 'client_credentials',
+        client_id: event.secrets.client_id,
+        client_secret: event.secrets.client_secret,
+        audience: event.secrets.audience
+      });
+  
+      try {
+        const tokenPromise = await fetch(event.secrets.gettokenpath, {
+            method: "POST",
+            headers: {
+              'Content-Type': "application/x-www-form-urlencoded"
+            },
+            body: params.toString()
+          });
+  
+        tokenResponse = await tokenPromise;
```

Comment:
Is there a need for await? Already const tokenPromise(line 39) is awaiting the fetch. The next steps will be executed only once the await fetch() is completed.